compiler/tools/fpp-depend/test/def_state_machine.fpp

```diff
@@ -0,0 +1,11 @@
+locate type A at "A.fpp"
+locate type G at "G.fpp"
+locate constant s at "s.fpp"
+
+state machine M {
+
+  action a: A
+  guard g: G
+  signal s: string size s
```

Comment:
Does it make sense to add warnings about unused definitions inside state machines?

Reply:
Yes, that makes sense. Right now we don't have any "warning report" capability for FPP, we just have detect-error-and-halt. We would need to design an interface to the warnings, e.g., can we turn them on and off, do they just display to the console, can they be promoted to errors?